Git/config.sublime-syntax

```diff
@@ -0,0 +1,163 @@
+      scope: support.constant.color-attribute.gitconfig  # TODO: correct scope?
+
+  color-hex:
+    - match: (#)\h{6}
+      captures:
+        0: constant.other.color.rgb-value.gitconfig
+        1: punctuation.definition.constant.gitconfig
+
+  color-named:
+    - match: \b(?:yellow|white|red|normal|magenta|green|cyan|blue|black|auto){{space_or_eol}}
+      scope: support.constant.color.gitconfig
+
+  comments:
+    - match: (#|;).*
+      captures:
+        1: punctuation.definition.comment.gitconfig
+      scope: comment.line.gitconfig
+
+  # '\b', '\n', '\t', '\"', '\\'
+  escape:
+    - match: \\[\\"bnt]
+      scope: constant.character.escape.gitconfig
+
+  illegal-newline:
+    - match: \n
+      scope: invalid.illegal.gitconfig
+      pop: true
+
+  # key = val
+  key-value-pair:
+    - match: ^\s*({{variable_name}})\s*(\=)
+      captures:
+        1: meta.key.gitconfig variable.language.gitconfig  # TODO: correct scope?
+        2: punctuation.separator.key-value.gitconfig
+      push:
+        # Hex colors like #efefef will be matched as comments unless we lower
+        # the precedence of comments. Here we disable the comments prototype and
+        # match comments after color-hex in the value rule.
+        - meta_include_prototype: false
+        - meta_scope: meta.key-value-pair.gitconfig
+        - meta_content_scope: meta.value.gitconfig
+        - match: '(?=^\s*{{variable_name}}\s*\=)'  # next key-value pair
+          pop: true
+        - match: (?=\[)                            # start of new section
+          pop: true
+        - include: value
+
+  # [section-name "subsection"]
+  section:
+    - match: \[
+      scope: punctuation.definition.section.begin.gitconfig
+      push:
+        - meta_scope: meta.brackets.gitconfig
+        - match: \]
+          scope: punctuation.definition.section.end.gitconfig
+          pop: true
+        - include: illegal-newline  # complain if '[' and ']' on separate lines
+        - include: section-name
+        - include: string-double    # subsection
+    - include: stray-bracket
+
+  section-name:
+    - match: '{{section_name}}'
+      scope: entity.name.gitconfig
+
+  stray-bracket:
+    - match: \]
+      scope: punctuation.definition.section.end.gitconfig invalid.illegal.stray-bracket.gitconfig
+
+  string:
+    - include: string-double-escaped-quotes
+    - include: string-double
+    - include: string-single-escaped-quotes
+    - include: string-single
+
+  string-double:
+    - match: '"'
+      scope: punctuation.definition.string.begin.gitconfig
+      push:
+        - meta_include_prototype: false
+        - meta_scope: string.quoted.double.gitconfig
+        - match: '"'
+          scope: punctuation.definition.string.end.gitconfig
+          pop: true
+        - include: escape
+
+  # some values use escaped quotes, e.g. --pretty=format:\"%h %ad | %s%d [%an]\"
+  string-double-escaped-quotes:
```

Comment:
Remove this context.

Reply:
Um, I'm not sure how to do that without increasing the potential for mismatches. With values (everything to the right of the `=` sign), I'm matching strings separately, at a high precedence. That way the text in those strings doesn't get mismatched as a color or boolean.
```
[foo]
    boolVar  = true  <- match as bool
    color    = blue  <- match as color
    bar      = --flag=\"true blue\"  <- don't want to match as bool or color
```

Reply:
OK, I see your point and I also see that it's been changed to an unquoted string so I'm happy. On the other hand, can something like this happen?

```
[foosection]
bar = \"  # a string of length one consisting of a double-quote character.
```